src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -538,8 +538,8 @@ public int available() throws IOException {
             if (available != 0) return available;
             Iterator<?> iterator = currentListItr;
             if (iterator != null && iterator.hasNext()) return 1;
-            if (buffers.isEmpty()) return 0;
-            return 1;
+            if (!buffers.isEmpty() && buffers.peek() != LAST_LIST ) return 1;
```

Comment:
Should probably be:

```
if (iterator != null && iterator.hasNext() && Utils.hasRemaining(buffers)) return 1;
```

Reply:
do you mean 

if (iterator != null && iterator.hasNext() && Utils.hasRemaining(buffers.peek())) return 1; ?

Reply:
Doh! Sorry ignore my suggestion. What you originally had seems reasonable. Give me some time to test.